### PR TITLE
Bh/code syntax highlighting

### PR DESCRIPTION
**Pull request by howebenhowe**
implements 1st pass on #799, to boot up a feature branch

_Test plan_

1. create a code block in a pub (by inserting ```)
2. as you write JS code, should hilight

### Still to implement

- UI for selection of various languages
- PDF export + SSR behavior
- possibly cache rendered HTML per node
- implement delayed client-side importing of selected language for syntax highlight
- default language selection

### CODE REVIEW

File: client/components/Editor/plugins/keymap.ts [hunk 1]
```diff
@@ -20,12 +20,29 @@ import { wrapInList, splitListItem, liftListItem, sinkListItem } from 'prosemirr
...
 import { undoInputRule } from 'prosemirror-inputrules';
 import { keymap } from 'prosemirror-keymap';
+import { Selection } from 'prosemirror-state';
 import { mathBackspaceCmd } from '@benrbray/prosemirror-math';
 
 import { splitBlockPreservingAttrs } from '../commands';
 
 const mac = typeof navigator !== 'undefined' ? /Mac/.test(navigator.platform) : false;
 
+const arrowHandler = (dir) => (state, dispatch, view) => {
```
Comment (idreyn) [inline client/components/Editor/plugins/keymap.ts:30]: Can we get some types on this bad boy?

**Review by idreyn — APPROVED** [review]
LGTM!

**Review by 3mcd — APPROVED** [review]
LGTM! It's nice to see that CodeMirror doesn't require you call a cleanup method for it to be garbage collected.

File: client/components/Editor/plugins/keymap.ts (hunk 1 above)
Comment (howebenhowe) [reply client/components/Editor/plugins/keymap.ts]: Resolved, I hope, by latest commit. Sometimes I fear over-typing, but I reckon that's misplaced fear.

File: client/components/Editor/plugins/keymap.ts [hunk 2]
```diff
@@ -18,14 +18,37 @@ import {
...
+	(state: EditorState, dispatch?: DispatchFn, view?: EditorView): boolean => {
+		if (state.selection.empty && view && view.endOfTextblock(dir)) {
+			const side = dir === 'left' || dir === 'up' ? -1 : 1;
+			const $head = state.selection.$head;
+			const nextPos = Selection.near(
+				state.doc.resolve(side > 0 ? $head.after() : $head.before()),
+				side,
+			);
+			if (nextPos.$head && nextPos.$head.parent.type.name === 'code_block') {
+				dispatch?.(state.tr.setSelection(nextPos));
```
Comment (howebenhowe) [inline client/components/Editor/plugins/keymap.ts:45]: Just learned that this conditional calling is possible. I'd seen it for conditional access of properties, but not considered that it's a terse `if` here, too.

**Pull request merged**